src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
Using a `StringType` will result in incorrect types when data is pushed into entities. Previously entities would have floats, if we used `StringType` they would have strings which could cause problems with application math. 

We could add a new DecimalType that adds the proposed casting and leave the FloatType alone. Having a separate type would allow us to only format non-string values with sprintf().